### PR TITLE
[PHP 8.2] Fixed some deprecation warnings

### PR DESCRIPTION
**Pull request by fballiano**
In https://github.com/OpenMage/magento-lts/issues/3152 @ADDISON74 found out some PHP 8.2 deprecation warnings, this PR should fix them.

Note: at the moment it still doesn't fix the problems with Mage_Paypal_Model_Config that apparently were introduced with https://github.com/OpenMage/magento-lts/pull/2759

### Related Pull Requests
1. https://github.com/OpenMage/magento-lts/pull/2759

### Fixed Issues
1. Fixes https://github.com/OpenMage/magento-lts/issues/3152

### CODE REVIEW

File: app/code/core/Mage/Rule/Model/Condition/Abstract.php [hunk 1]
```diff
@@ -812,6 +812,7 @@ public function validateAttribute($validatedValue)
      */
     protected function _compareValues($validatedValue, $value, $strict = true)
     {
+        $validatedValue = $validatedValue ?? '';
         if ($strict && is_numeric($validatedValue) && is_numeric($value)) {
             return $validatedValue == $value;
         } else {
```
Comment (kiatng) [inline app/code/core/Mage/Rule/Model/Condition/Abstract.php:818]:
```suggestion
        if ($strict && is_numeric($validatedValue) && is_numeric($value)) {
            return $validatedValue == $value;
        } else {
             $validatedValue = $validatedValue ?? '';
```

File: app/code/core/Mage/Eav/Model/Entity/Attribute/Source/Table.php [hunk 2]
```diff
@@ -75,7 +75,7 @@ public function getAllOptions($withEmpty = true, $defaultValues = false)
     public function getOptionText($value)
     {
         $isMultiple = false;
-        if (strpos($value, ',')) {
+        if (strpos($value ?? '', ',')) {
```
Comment (kiatng) [inline app/code/core/Mage/Eav/Model/Entity/Attribute/Source/Table.php:78]:
```suggestion
        if ($value && strpos($value, ',')) {
```

File: app/code/core/Mage/Rule/Model/Condition/Abstract.php (hunk 1 above)
Comment (fballiano) [reply app/code/core/Mage/Rule/Model/Condition/Abstract.php]: it should be ok now :-)

File: app/code/core/Mage/SalesRule/Model/Validator.php [hunk 3]
```diff
@@ -962,7 +962,7 @@ protected function _addDiscountDescription($address, $rule)
         $label = '';
         if ($ruleLabel) {
             $label = $ruleLabel;
-        } elseif (strlen($address->getCouponCode())) {
+        } elseif (strlen($address->getCouponCode() ?? '')) {
```
Comment (elidrissidev) [inline app/code/core/Mage/SalesRule/Model/Validator.php:965]: `coupon_code` getter in Quote model was overridden in #2825 to always return empty string, maybe a similar approach can be done with address.

Comment (fballiano) [reply app/code/core/Mage/SalesRule/Model/Validator.php]: @elidrissidev I think I did it the right way now

File: app/code/core/Mage/ConfigurableSwatches/Helper/Data.php [hunk 4]
```diff
@@ -75,6 +75,7 @@ public function getHyphenatedString($str)
      */
     public static function normalizeKey($key)
     {
+        if ($key === null || $key === '') return '';
```
Comment (kiatng) [inline app/code/core/Mage/ConfigurableSwatches/Helper/Data.php:78]:
```suggestion
        if ($key === null || $key === '') {
             return '';
        }
```

File: app/code/core/Mage/ConfigurableSwatches/Helper/Data.php [hunk 5]
```diff
@@ -75,6 +75,9 @@ public function getHyphenatedString($str)
      */
     public static function normalizeKey($key)
     {
+        if ($key === null || $key === '') {
+             return '';
```
Comment (fballiano) [inline app/code/core/Mage/ConfigurableSwatches/Helper/Data.php:79]:
```suggestion
            return '';
```

**Review by kiatng — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]
Let's merge it and I will make a list with the warnings left behind we still get in the log files.

**Pull request merged**